### PR TITLE
[Ready to merge] add --gateway to store deploy command & rename shorthand -g to -s for store list and store inspect

### LINKED ISSUE #315: 0.6.2: `faas-cli store deploy function --gateway` gives error
## Expected Behaviour

Expected: you should be able to specify --gateway to point your faas-cli at a local gateway for all `faas-cli store` operations.

## Current Behaviour

This change was introduced in 0.6.2:

https://github.com/openfaas/faas-cli/commit/9abc0880aa84cd1497a50ea2d55eb4af6fc38422#diff-4d8c2dcb33807ea71d97dffb92481be6

It removes the `-g` flag that had pointed to the store, replacing it with a `-u` flag, but not restoring the gateway function.

## Possible Solution

Put gateway logic into store function.

## Steps to Reproduce (for bugs)
1. using `brew` install version 0.6.2

```
% faas-cli store deploy Colorise --gateway http://127.0.0.1:31112/
Unknown flag: --gateway
```

## Context

Workaround is deploying function via UI, but that doesn't let me set environment variables.

## Your Environment
* Docker version `docker version` (e.g. Docker 17.0.05 ):

Docker 18.02.0-ce-rc2 darwin/amd64

* Are you using Docker Swarm or Kubernetes (FaaS-netes)?

FaaS-netes

* Operating System and version (e.g. Linux, Windows, MacOS):

Darwin Eds-MacBook-Pro.attlocal.net 16.7.0 Darwin Kernel Version 16.7.0: Wed Oct  4 00:17:00 PDT 2017; root:xnu-3789.71.6~1/RELEASE_X86_64 x86_64

* Link to your project or a code example to reproduce issue:

### PR DESCRIPTION
**Pull request by itscaro**
## Description
Rename `-g` to `-u` in `store list` & `store inspect` commands to avoid mistake when using shorthand.

Add `--gateway` `-g` and `handler` to `store deploy`

Remove `--handler` and `--lang` from usage help

```
faas-cli -g localhost:31112 store list
Cannot connect to OpenFaaS store on URL: localhost:31112
```

```
faas-cli store deploy Left-Pad -g 127.0.0.1:31112
Unknown shorthand flag: 'g' in -g
```

```
faas-cli store deploy Left-Pad --gateway 127.0.0.1:31112
Unknown flag: --gateway
```

## Motivation and Context

- [x] issue to propose this change (Fixes #315)

## How Has This Been Tested?
````
faas-cli fix-gateway-flag *$% = $ .\faas-cli.exe store deploy left-pad -g http://localhost:8080

Deployed. 200 OK.
URL: http://localhost:8080/function/leftpad
```

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks @itscaro that was really fast 🐎 🏎 . I think I got some of the changes in at a similar time before I realised you were also addressing this. Can you rebase on master and see if there's any changes left to merge?

File: commands/store.go [hunk 1]
```diff
@@ -33,14 +33,16 @@ const (
 func init() {
 	// Setup flags that are used by multiple commands (variables defined in faas.go)
 	storeCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
+	storeCmd.PersistentFlags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative URL starting with http(s)://")
```
Comment (itscaro) [inline commands/store.go]: Since 3 sub commands need this flag, better to make it persistent to the namespace `store`

File: commands/store.go [hunk 2]
```diff
@@ -33,14 +33,16 @@ const (
...
 
 	// Setup flags used by store command
-	storeListCmd.Flags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative URL starting with http(s)://")
 	storeListCmd.Flags().BoolVarP(&verboseDescription, "verbose", "v", false, "Verbose output for the field values")
 
-	storeInspectCmd.Flags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative Store URL starting with http(s)://")
 	storeInspectCmd.Flags().BoolVarP(&verboseDescription, "verbose", "v", false, "Verbose output for the field values")
 
+	// Setup flags that are used by multiple commands (variables defined in faas.go)
```
Comment (itscaro) [inline commands/store.go]: add `gateway` and `network` flags

File: commands/store.go [hunk 3]
```diff
@@ -94,7 +97,7 @@ var storeDeployCmd = &cobra.Command{
 	Long:  `Same as faas-cli deploy except pre-loaded with arguments from the store`,
 	Example: `  faas-cli store deploy figlet
   faas-cli store deploy figlet \
-    --gateway=http://localhost:8080 --lang=python \
+    --gateway=http://localhost:8080 \
```
Comment (itscaro) [inline commands/store.go]: --lang is unused

File: commands/store.go [hunk 4]
```diff
@@ -33,14 +33,16 @@ const (
...
 
 	// Setup flags used by store command
-	storeListCmd.Flags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative URL starting with http(s)://")
 	storeListCmd.Flags().BoolVarP(&verboseDescription, "verbose", "v", false, "Verbose output for the field values")
 
-	storeInspectCmd.Flags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative Store URL starting with http(s)://")
 	storeInspectCmd.Flags().BoolVarP(&verboseDescription, "verbose", "v", false, "Verbose output for the field values")
 
+	// Setup flags that are used by multiple commands (variables defined in faas.go)
+	storeDeployCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
```
Comment (amirkarimi) [inline commands/store.go]: Just for my knowledge, why deploy command needs this flag explicitly but for `list` and `inspect` the [parent command flag](https://github.com/openfaas/faas-cli/pull/309/files#diff-4d8c2dcb33807ea71d97dffb92481be6R35) is enough?

Comment (rgee0) [reply commands/store.go]: The commands are heading off in different directions.  list and inspect are hitting GitHub (or a provided URI) for JSON file and printing out detail contained in that JSON file.  Deploy is providing detail into a gateway (remote if provided, localhost if not) for the application to act upon.

Comment (amirkarimi) [reply commands/store.go]: You're right. I was thinking this PR tries to make `gateway` flag available for all sub-commands which was not needed for list and inspect.

Comment (amirkarimi) [reply commands/store.go]: Plus, we can get rid of [gateway flag on store command](https://github.com/openfaas/faas-cli/pull/309/files#diff-4d8c2dcb33807ea71d97dffb92481be6R35). @itscaro

**Comment by alexellis** [comment]
@itscaro we have a conflict preventing merge on the following file:

* commands/store.go

Please could you rebase to resolve the merge issue and then squash all the commits into one signed commit?

**Comment by itscaro** [comment]
@alexellis rebased, squashed and good to merge 👍

**Comment by alexellis** [comment]
Merged. Thank you @itscaro for seeing this one through. 👍

**Pull request merged**